ArduCopter/system.cpp

```diff
@@ -516,6 +516,41 @@ const char* Copter::get_frame_string()
     }
 }
 
+// return string corresponding to frame_type
+const char* Copter::get_type_string()
+{
+    switch (g.frame_type) {
+    case 0:
+        return "Plus";
+    case 1:
+        return "X";
+    case 2:
+        return "V";
+    case 3:
+        return "H";
+    case 4:
+        return "V-Tail";
+    case 5:
+        return "A-Tail";
+    case 10:
+        return "Y6B";
+    case 11:
+        return "Y6F";
+    case 12:
+        return "BetaFlightX";
+    case 13:
+        return "DJIX";
+    case 14:
+        return "ClockwiseX";
+    case 15:
+        return "I";
+    case 18:
+        return "BetaFlightXReversed";
+    default:
```

Comment:
of you remove default this is more likely to be kept up to date

Reply:
you will have to cast to the enum for that to work

Reply:
The FRAME_TYPE value does not have a range.
Undefined values can be set by GCS.
I was able to verify this with APM PLANNER2.
I would modify it to show the set value if an undefined value is set.

![Screenshot from 2020-08-31 07-13-41](https://user-images.githubusercontent.com/646194/91670596-0fa0c580-eb5a-11ea-9c2f-1a140b0caec7.png)

Reply:
@muramura `enum class` required at the least.  Handling the case where the parameter is out-of-range can be done *outside* the switch statement, allowing you to have no `default:` clause (ensuring someone adding a new frame must add the string.  The existing equivalent code for the mav-type is not good.


What's the byte-cost of this new code?

The function name also needs to be more specific.

Reply:
I have defined the frame type in ENUM.

Reply:
The type of the frame type variable is AP_Int8.
I don't think you can specify the type of ENUM in the config parameter.
I can set it to a value other than the ENUM value.
I was able to set it in APM PLANNER2.
I think defult is valid.

Reply:
> The type of the frame_type variable is AP_Int8.

Yes.

> I don't think you can specify the type of ENUM in the config parameter.

You cast it as part of the switch statement.

> I can set it to a value other than the ENUM value.

Yes, you can.

> I think defult is valid.

No, it is not.  We use the switch statement to ensure the programmer takes care of adding the frame string.

If the value is out of range then you handle that after the body of the switch statement by returning your "DEFAULT" string.

Reply:
Displaying the frame type can reduce the time required for the following steps.
1. start up the GCS.
2 Select the configuration parameter item with the mouse.
3 Key in FRAME_TYPE.

Reply:
We already have a enum https://github.com/ArduPilot/ardupilot/blob/abee62abc453da8f558cbd76d34ce7b301a2e6ce/libraries/AP_Motors/AP_Motors_Class.h#L49

Reply:
@IamPete1 san. I'll change it to motor_frame_type.